### ISSUE TITLE
Missing width and height attributes in svg document lead to crash

### ISSUE THREAD

**Issue by Pierre-Andre** [textext/textext#19]
- When using TeXtext on an empty document with $\beta_1$ TeX text it works (obviously).
- But when I use texText on a more complicated svg (created whith R and svglite device) the same textext object throw the following error:
```
--------------------------------------------------
expected string or buffer
Traceback (most recent call last):
  File "/home/pac/.config/inkscape/extensions/asktext.py", line 491, in cb_ok
    self.callback(self.text, self.preamble_file, self.global_scale_factor)
  File "textext.py", line 224, in <lambda>
    old_node),
  File "textext.py", line 351, in do_convert
    width = self.unittouu(root.get('width'))
  File "/usr/share/inkscape/extensions/inkex.py", line 387, in unittouu
    p = param.match(string)
TypeError: expected string or buffer
----------------------------------------------------
```
I do suspect a problem with getting the size of the page (and may be a default/fallback width can be useful ?)

**Comment by leberwurstsaft** [comment]
Good point. Width and height are not required values and are implicitly assumed to be "100%" when missing. Question is, what would 100% refer to when it is the rood node?

The recommended way to specify dimensions according to some article I read is to define a viewBox anyway. So we could also look at the viewBox, either as a fallback or as the primary way to retrieve the doc's dimensions. It looks like in the example by @sizmailov the viewBox uses document units rather than mm or anything from the real word?

**Comment by Pierre-Andre** [comment]
The svg is not created by inkscape and it does not provide height and width. It does provide a viewBox (in pixels, even it is not explicitely written in svg files, inkscape understand it as pixels):

```<?xml version='1.0' encoding='UTF-8' ?>
<svg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 198.43 198.43'>
<defs>
  <style type='text/css'><![CDATA[
    line, polyline, polygon, path, rect, circle {
      fill: none;
      stroke: #000000;
      stroke-linecap: round;
      stroke-linejoin: round;
      stroke-miterlimit: 10.00;
    }
  ]]></style>
</defs>
<rect width='100%' height='100%' style='stroke: none; fill: #FFFFFF;'/>
<defs>
'''
Of course, when I modify the document properties (ctrl+maj-d) I can resize page dimension to content (which insert height and width in svg) and TeXtext will work as desired, but I do think that it should work with viewbox only...

**Comment by jcwinkler** [comment]
OK, coming back to this issue. 

Instead of
``` python
width = self.unittouu(root.get('width'))
height = self.unittouu(root.get('height'))
```
we could just write
``` python
width = self.unittouu(self.getDocumentWidth())
height = self.unittouu(self.getDocumentHeight())
```

`inkex.getDocumentHeight()` falls internally back to to viewbox of no width and height attributes are found. If even no viewbox attribute is found 0 is returned. But then we would be lost, anyway, I guess.

**Comment by jcwinkler** [comment]
( inkex does not provide getDocumentWidth() in Inkscape 0.48.x ). If we want to keep compatibility with this version...

**Comment by sizmailov** [comment]
If I recall it right width/height only affect where fresh node would appear. It's reasonable to try place it at center of document or as a fallback at left top corner. 

The pythonic way to handle it would be: 

```python
try:
    try:  # try to use viewbox first
        width = self.unittouu(self.getDocumentWidth())
        height = self.unittouu(self.getDocumentHeight())
    except (AttributeError,TypeError,ValueError):  # look for width/height attributes in root
        width = self.unittouu(root.get('width'))
        height = self.unittouu(root.get('height'))
except (AttributeError,TypeError,ValueError):  # fallback 
    width = 0
    height = 0
```

**Comment by sizmailov** [comment]
In case of fallback we can also inform user that new node is going to be created at (0,0) in the bottom of TexText window.

**Comment by jcwinkler** [comment]
> except (AttributeError,TypeError,ValueError): # look for width/height attributes in root 
> width = self.unittouu(root.get('width')) 
> height = self.unittouu(root.get('height'))

isn't this already done in getDocumentWidth ?

Final fallback for placement (0,0) is a good idea!

**Comment by jcwinkler** [comment]
Fixed in release 0.7.3
